### PR TITLE
Dashboard: extend drill events

### PR DESCRIPTION
**Pull request by kandl**
- Notify about start of the drill resolution
- These events are necessary to cover KD functionality, where drill to custom url opens window immediately and then do the redirect once url is resolved
- Slightly change the naming of drill events

JIRA: RAIL-3385

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-ui-dashboard/src/model/events/index.ts [hunk 1]
```diff
@@ -135,20 +142,34 @@ export {
...
+    DashboardDrillToCustomUrlResolved,
+    DashboardDrillToDashboardResolved,
+    DashboardDrillToInsightResolved,
+    DashboardDrillToLegacyDashboardResolved,
+    drillDownResolved,
+    drillToAttributeUrlResolved,
+    drillToCustomUrlResolved,
+    drillToDashboardResolved,
+    drillToInsightResolved,
+    drillToLegacyDashboardResolved, // DashboardDrillResolved, // drillResolved,
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/events/index.ts:172]: Maybe a forgotten comment?

File: libs/sdk-ui-dashboard/src/model/events/drill.ts [hunk 2]
```diff
@@ -50,14 +50,65 @@ export interface DashboardDrillTriggered extends IDashboardEvent {
...
+        correlationId,
+        payload: {
+            drillEvent,
+            drillContext,
+        },
+    };
+}
+
+/**
+ * A general drill event that is emitted each time any dashboard drill is requested.
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/events/drill.ts:71]: The comment talks about event being requested, but it is called resolved. Is this intentional?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4632/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6257/ : SUCCESS in 20m 06s

File: libs/sdk-ui-dashboard/src/model/events/drill.ts (hunk 2 above)
Comment (kandl) [reply libs/sdk-ui-dashboard/src/model/events/drill.ts:71]: Good catch, I'll correct it.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1582

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1311/ : SUCCESS in 1m 15s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1587/ : SUCCESS in 23s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/361/ : SUCCESS in 13m 00s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6259/ : SUCCESS in 17m 30s

**Pull request merged**